runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcCommitWorkStream.java

```diff
@@ -156,29 +163,44 @@ public void sendHealthCheck() {
   protected void onResponse(StreamingCommitResponse response) {
     commitWorkThrottleTimer.stop();
 
-    RuntimeException finalException = null;
+    CommitCompletionException failures = new CommitCompletionException();
     for (int i = 0; i < response.getRequestIdCount(); ++i) {
       long requestId = response.getRequestId(i);
       if (requestId == HEARTBEAT_REQUEST_ID) {
         continue;
       }
-      PendingRequest done = pending.remove(requestId);
-      if (done == null) {
-        LOG.error("Got unknown commit request ID: {}", requestId);
+      PendingRequest pendingRequest = pending.remove(requestId);
+      CommitStatus commitStatus =
+          i < response.getStatusCount() ? response.getStatus(i) : CommitStatus.OK;
+      if (pendingRequest == null) {
+        if (!hasReceivedShutdownSignal()) {
+          // Skip responses when the stream is shutdown since they are now invalid.
```

Comment:
nit: skip responses is unclear, maybe somethign like 
 // Missing responses is expected after shutdown because it removes them.

Reply:
done